From 4f327f9d32e7366d6929bc49ef2e7917adcb8c04 Mon Sep 17 00:00:00 2001
From: Howard Hsu <howard-yh.hsu@mediatek.com>
Date: Fri, 2 May 2025 10:48:23 +0800
Subject: [PATCH 3/3] mtk: mt76: mt7996: fix 4 byte alignment for EDCCA command

The mt7992 firmware will skip to run mcu cmd but still reply the response
to driver if the mcu cmd is not 4 byte alignment. Fix the alignment of
mcu cmd 'UNI_BAND_CONFIG_EDCCA_THRESHOLD'.

Signed-off-by: Howard Hsu <howard-yh.hsu@mediatek.com>

---
 mt7996/mtk_mcu.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mt7996/mtk_mcu.c b/mt7996/mtk_mcu.c
index 72e5fa28..e610e262 100644
--- a/mt7996/mtk_mcu.c
+++ b/mt7996/mtk_mcu.c
@@ -107,6 +107,7 @@ int mt7996_mcu_edcca_threshold_ctrl(struct mt7996_phy *phy, u8 *value, bool set)
 		__le16 len;
 		u8 threshold[4];
 		bool init;
+		u8 _rsv2[3];
 	} __packed *res, req = {
 		.band_idx = phy->mt76->band_idx,
 		.tag = cpu_to_le16(UNI_BAND_CONFIG_EDCCA_THRESHOLD),
-- 
2.45.2

